apps/backend/src/modules/projects/project.service.ts

```diff
@@ -14,10 +15,30 @@ import { ProjectEntity } from "./project.entity.js";
 import { type ProjectRepository } from "./project.repository.js";
 
 class ProjectService implements Service {
+	private projectApiKeyService: ProjectApiKeyService;
+
 	private projectRepository: ProjectRepository;
 
-	public constructor(projectRepository: ProjectRepository) {
+	public constructor(
+		projectRepository: ProjectRepository,
+		projectApiKeyService: ProjectApiKeyService,
+	) {
 		this.projectRepository = projectRepository;
+		this.projectApiKeyService = projectApiKeyService;
+	}
+
+	private async injectProjectApiKeyToProject(
+		project: ProjectGetAllItemResponseDto,
+	): Promise<ProjectGetAllItemResponseDto> {
+		const projectApiKey = await this.projectApiKeyService.findByProjectId(
+			project.id,
+		);
+		const apiKey = projectApiKey ? projectApiKey.apiKey : null;
+
+		return {
+			...project,
+			apiKey,
+		};
 	}
```

Comment:
let's do this separately for each method. We could move it to a separate method if there was more then one call, but it definitely wouldn't be something like inject, it would be just - getProjectApiKey
```suggestion
```